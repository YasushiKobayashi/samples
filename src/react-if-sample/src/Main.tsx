import * as React from 'react'

export const Bad: React.FC<{ isAdmin: boolean }> = ({ isAdmin }) => {
  return (
    <>
      {isAdmin && <p>adminにはこれを見せるよ</p>}
      {!isAdmin && <p>userにはこっち見せるよ</p>}
      <p>両方に見せるよ</p>
      {isAdmin && <p>adminだけこの情報も見せるよ</p>}
    </>
  )
}

export const Better: React.FC<{ isAdmin: boolean }> = ({ isAdmin }) => {
  if (isAdmin) {
    return (
      <>
        <p>adminにはこれを見せるよ</p>
        <p>両方に見せるよ</p>
        <p>adminだけこの情報も見せるよ</p>
      </>
    )
  }

  return (
    <>
      <p>userにはこっち見せるよ</p>
      <p>両方に見せるよ</p>
    </>
  )
}

const AdminComponent = () => {
  return (
    <>
      <p>adminにはこれを見せるよ</p>
      <p>両方に見せるよ</p>
      <p>adminだけこの情報も見せるよ</p>
    </>
  )
}

const UserComponent = () => {
  return (
    <>
      <p>userにはこっち見せるよ</p>
      <p>両方に見せるよ</p>
    </>
  )
}

export const Good: React.FC<{ isAdmin: boolean }> = ({ isAdmin }) => {
  if (isAdmin) {
    return <AdminComponent />
  }

  return <UserComponent />
}
