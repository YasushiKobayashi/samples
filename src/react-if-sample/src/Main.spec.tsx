import * as React from 'react'
import { cleanup, render } from '@testing-library/react'

import { Bad, Better, Good } from './Main'

describe('Main', () => {
  afterEach(() => {
    cleanup()
  })
  it('Bad admin', () => {
    const { container, asFragment } = render(<Bad isAdmin />)
    expect(container.textContent).toContain('adminにはこれを見せるよ')
    expect(container.textContent).toContain('adminだけこの情報も見せるよ')
    expect(container.textContent).not.toContain('userにはこっち見せるよ')
    expect(asFragment()).toMatchSnapshot()
  })

  it('Bad user', () => {
    const { container, asFragment } = render(<Bad isAdmin={false} />)
    expect(container.textContent).not.toContain('adminにはこれを見せるよ')
    expect(container.textContent).not.toContain('adminだけこの情報も見せるよ')
    expect(container.textContent).toContain('userにはこっち見せるよ')
    expect(asFragment()).toMatchSnapshot()
  })

  it('Better admin', () => {
    const { container, asFragment } = render(<Better isAdmin />)
    expect(container.textContent).toContain('adminにはこれを見せるよ')
    expect(container.textContent).not.toContain('userにはこっち見せるよ')
    expect(asFragment()).toMatchSnapshot()
  })

  it('Better user', () => {
    const { container, asFragment } = render(<Better isAdmin={false} />)
    expect(container.textContent).not.toContain('adminにはこれを見せるよ')
    expect(container.textContent).toContain('userにはこっち見せるよ')
    expect(asFragment()).toMatchSnapshot()
  })

  it('Good admin', () => {
    const { container, asFragment } = render(<Good isAdmin />)
    expect(container.textContent).toContain('adminにはこれを見せるよ')
    expect(container.textContent).not.toContain('userにはこっち見せるよ')
    expect(asFragment()).toMatchSnapshot()
  })

  it('Good user', () => {
    const { container, asFragment } = render(<Good isAdmin={false} />)
    expect(container.textContent).not.toContain('adminにはこれを見せるよ')
    expect(container.textContent).toContain('userにはこっち見せるよ')
    expect(asFragment()).toMatchSnapshot()
  })
})
