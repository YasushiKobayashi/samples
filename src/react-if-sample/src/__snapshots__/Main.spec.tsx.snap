// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Bad admin 1`] = `
<DocumentFragment>
  <p>
    adminにはこれを見せるよ
  </p>
  <p>
    両方に見せるよ
  </p>
  <p>
    adminだけこの情報も見せるよ
  </p>
</DocumentFragment>
`;

exports[`Main Bad user 1`] = `
<DocumentFragment>
  <p>
    userにはこっち見せるよ
  </p>
  <p>
    両方に見せるよ
  </p>
</DocumentFragment>
`;

exports[`Main Better admin 1`] = `
<DocumentFragment>
  <p>
    adminにはこれを見せるよ
  </p>
  <p>
    両方に見せるよ
  </p>
  <p>
    adminだけこの情報も見せるよ
  </p>
</DocumentFragment>
`;

exports[`Main Better user 1`] = `
<DocumentFragment>
  <p>
    userにはこっち見せるよ
  </p>
  <p>
    両方に見せるよ
  </p>
</DocumentFragment>
`;

exports[`Main Good admin 1`] = `
<DocumentFragment>
  <p>
    adminにはこれを見せるよ
  </p>
  <p>
    両方に見せるよ
  </p>
  <p>
    adminだけこの情報も見せるよ
  </p>
</DocumentFragment>
`;

exports[`Main Good user 1`] = `
<DocumentFragment>
  <p>
    userにはこっち見せるよ
  </p>
  <p>
    両方に見せるよ
  </p>
</DocumentFragment>
`;
